homeassistant/components/device_tracker/traccar.py

```diff
@@ -0,0 +1,95 @@
+"""
+Support for Traccar device tracking.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/device_tracker.traccar/
+"""
+from datetime import timedelta
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components.device_tracker import PLATFORM_SCHEMA
+from homeassistant.const import (
+    CONF_HOST, CONF_PORT, CONF_SSL, CONF_VERIFY_SSL,
+    CONF_PASSWORD, CONF_USERNAME)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.event import async_track_time_interval
+from homeassistant.util import slugify
+
+
+REQUIREMENTS = ['pytraccar==0.1.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_ADDRESS = 'address'
+ATTR_CATEGORY = 'category'
+ATTR_GEOFENCE = 'geofence'
+ATTR_TRACKER = 'tracker'
+
+DEFAULT_SCAN_INTERVAL = timedelta(seconds=30)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=8082): cv.port,
+    vol.Optional(CONF_SSL, default=False): cv.boolean,
```

Comment:
I would suggest that we are using a secured channel for the communication by default, if possible.

Reply:
It is possible, but I think tha would confuse the potential enduser?
Home Assistant itself and most of the components/platforms do have this as false by defaul?
```bash
git grep "CONF_SSL, default=True"
homeassistant/components/sensor/synologydsm.py:    vol.Optional(CONF_SSL, default=True): cv.boolean,
```

```bash
git grep "CONF_SSL, default=False"
homeassistant/components/binary_sensor/hikvision.py:    vol.Optional(CONF_SSL, default=False): cv.boolean,
homeassistant/components/device_tracker/netgear.py:    vol.Optional(CONF_SSL, default=False): cv.boolean,
homeassistant/components/device_tracker/tomato.py:    vol.Optional(CONF_SSL, default=False): cv.boolean,
homeassistant/components/device_tracker/traccar.py:    vol.Optional(CONF_SSL, default=False): cv.boolean,
homeassistant/components/media_player/epson.py:    vol.Optional(CONF_SSL, default=False): cv.boolean,
homeassistant/components/octoprint.py:        vol.Optional(CONF_SSL, default=False): cv.boolean,
homeassistant/components/sensor/nzbget.py:    vol.Optional(CONF_SSL, default=False): cv.boolean,
homeassistant/components/sensor/pyload.py:    vol.Optional(CONF_SSL, default=False): cv.boolean,
homeassistant/components/sensor/qnap.py:    vol.Optional(CONF_SSL, default=False): cv.boolean,
homeassistant/components/sensor/radarr.py:    vol.Optional(CONF_SSL, default=False): cv.boolean,
homeassistant/components/sensor/sma.py:    vol.Optional(CONF_SSL, default=False): cv.boolean,
homeassistant/components/sensor/sonarr.py:    vol.Optional(CONF_SSL, default=False): cv.boolean,
homeassistant/components/splunk.py:        vol.Optional(CONF_SSL, default=False): cv.boolean,
```
```bash
git grep "DEFAULT_SSL ="
homeassistant/components/binary_sensor/concord232.py:DEFAULT_SSL = False
homeassistant/components/binary_sensor/nx584.py:DEFAULT_SSL = False
homeassistant/components/climate/venstar.py:DEFAULT_SSL = False
homeassistant/components/device_tracker/ddwrt.py:DEFAULT_SSL = False
homeassistant/components/device_tracker/luci.py:DEFAULT_SSL = False
homeassistant/components/media_player/emby.py:DEFAULT_SSL = False
homeassistant/components/media_player/firetv.py:DEFAULT_SSL = False
homeassistant/components/media_player/itunes.py:DEFAULT_SSL = False
homeassistant/components/rainmachine/__init__.py:DEFAULT_SSL = True
homeassistant/components/sabnzbd.py:DEFAULT_SSL = False
homeassistant/components/sensor/influxdb.py:DEFAULT_SSL = False
homeassistant/components/sensor/mfi.py:DEFAULT_SSL = True
homeassistant/components/sensor/pi_hole.py:DEFAULT_SSL = False
homeassistant/components/sensor/plex.py:DEFAULT_SSL = False
homeassistant/components/sensor/tautulli.py:DEFAULT_SSL = False
homeassistant/components/splunk.py:DEFAULT_SSL = False
homeassistant/components/switch/mfi.py:DEFAULT_SSL = True
homeassistant/components/zabbix.py:DEFAULT_SSL = False
homeassistant/components/zoneminder/__init__.py:DEFAULT_SSL = False
```

Reply:
I do not disagree with the suggestion, but to my knowledge, there is not a standard that a component or a platform should have that enabled by default, it may be something to discuss in <https://github.com/home-assistant/architecture>?

Reply:
I'm not sure if their paid service is running SSL, but the demo versions are not.
They do not have built-in support for SSL, you would have to run it behind a proxy to get that for a local instance.
I have used a local instance (hassio add-on) while building and testing this.